bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FileInputImplV2.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * ~ Copyright 2023 Adobe
+ * ~
+ * ~ Licensed under the Apache License, Version 2.0 (the "License");
+ * ~ you may not use this file except in compliance with the License.
+ * ~ You may obtain a copy of the License at
+ * ~
+ * ~ http://www.apache.org/licenses/LICENSE-2.0
+ * ~
+ * ~ Unless required by applicable law or agreed to in writing, software
+ * ~ distributed under the License is distributed on an "AS IS" BASIS,
+ * ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * ~ See the License for the specific language governing permissions and
+ * ~ limitations under the License.
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+package com.adobe.cq.forms.core.components.internal.models.v2.form;
+
+import java.util.Map;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Default;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.jetbrains.annotations.NotNull;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.forms.core.components.internal.form.FormConstants;
+import com.adobe.cq.forms.core.components.internal.models.v1.form.FileInputImpl;
+import com.adobe.cq.forms.core.components.models.form.FileInput;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { FileInput.class,
+        ComponentExporter.class },
+    resourceType = { FormConstants.RT_FD_FORM_FILE_INPUT_V2 })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class FileInputImplV2 extends FileInputImpl {
```

Comment:
implements FileInput

Reply:
I need fileinputimpl with one extra property that why extended the FileInputImpl. Should I write whole FileInputImplV2 similar to FileInputImpl by implementing FileInput and then adding my properties above that?

Reply:
You need to extend the base implementation, but adding an explicit implements makes the code more readable

Reply:
CWIIW but as FileInputImplV2 is extending the FileInputImpl which is already implementing the FileInput hence its not requried to implement in V2 as well.